mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -654,6 +668,95 @@ public void onClick(View view) {
         }
     }
 
+    public static class HabpanelFragment extends Fragment implements
+            ConnectionFactory.UpdateListener {
+        Connection mConnection;
+
+        public static HabpanelFragment newInstance() {
+            return new HabpanelFragment();
+        }
+
+        @Override
+        public View onCreateView(@NonNull LayoutInflater inflater,
+                ViewGroup container, Bundle savedInstanceState) {
+            View view = inflater.inflate(R.layout.fragment_habpanel, container, false);
+
+            TextView retryButton = view.findViewById(R.id.retry_button);
+            retryButton.setOnClickListener(v -> loadHabpanel(view));
+            loadHabpanel(view);
+
+            return view;
+        }
+
+        private void loadHabpanel(View view) {
+            try {
+                mConnection = ConnectionFactory.getUsableConnection();
+            } catch (ConnectionException e) {
+                updateViewVisibility(true, false, view);
+                return;
+            }
+
+            if (mConnection == null) {
+                updateViewVisibility(true, false, view);
+                return;
+            }
+            updateViewVisibility(false, true, view);
+
+            String url = mConnection.getAsyncHttpClient().buildUrl("/habpanel/index.html")
+                    .toString();
+
+            WebView webView = view.findViewById(R.id.webview);
+
+            webView.setWebViewClient(new AnchorWebViewClient(url,
+                    mConnection.getUsername(), mConnection.getPassword()) {
+                @Override
+                public void onPageFinished(WebView view, String url) {
+                    updateViewVisibility(false, false, getView());
+                }
+
+                @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+                @Override
+                public void onReceivedError(WebView view, WebResourceRequest request,
+                        WebResourceError error) {
+                    String url = request.getUrl().toString();
+                    Log.e(TAG, "onReceivedError() on URL: " + url);
+                    if (url.endsWith("/rest/events")) {
+                        updateViewVisibility(true, false, getView());
+                    }
+                }
+
+                @Override
+                public void onReceivedError(WebView view, int errorCode, String description,
+                        String failingUrl) {
+                    Log.e(TAG, "onReceivedError() (old) on URL: " + failingUrl);
+                    updateViewVisibility(true, false, getView());
```

Comment:
Why don't you check the failing URL here like you do above?

Reply:
https://github.com/openhab/openhab-android/pull/1124#discussion_r241302142
This is the deprecated version, so it's only called for the main page, right?

Reply:
Just re-checked the docs; you're right, so never mind.